### DISCUSSION TITLE
should connection config be symmetric ?

### DISCUSSION THREAD

**Discussion by hcojo** [strongswan/strongswan#1042] [Q&A]
hi
in general - should connection config be symmetric? 
and specifically - start_action? 

**can we have this configuration on one node:**

connections {
    pok1-qz1-sr1-rk025-s17-10-66-25-35-ipv4 {
        local_addrs = pok1-qz1-sr1-rk003-s17.ipv4.vcn.vpc.cloud.ibm.local
        remote_addrs = pok1-qz1-sr1-rk025-s17.ipv4.vcn.vpc.cloud.ibm.local
        dpd_delay = 2s
        local {
            auth = psk
            # no cert required for psk
        }
        remote {
            auth = psk
        }
        children {
            pok1-qz1-sr1-rk025-s17-10-66-25-35-ipv4 {
                local_ts = 11.0.209.0/24
                remote_ts = 11.6.81.0/24
                mode = tunnel
                inactivity = 0
                start_action = start
                dpd_action = restart
                esp_proposals = aes256gcm8-sha256-modp2048
                rekey_time=60m
                rand_time=6m
                interface=ipsec0
                mark_in=5
            }
        }
        version = 2
        proposals = aes256gcm8-sha256-modp2048
        rekey_time=240m
        rand_time=24m
    }
}

**and this one in the second node:** 

connections {
    pok1-qz1-sr1-rk003-s17-10-66-3-35-ipv4 {
        local_addrs = pok1-qz1-sr1-rk025-s17.ipv4.vcn.vpc.cloud.ibm.local
        remote_addrs = pok1-qz1-sr1-rk003-s17.ipv4.vcn.vpc.cloud.ibm.local
        dpd_delay = 2s
        local {
            auth = psk
            # no cert required for psk
        }
        remote {
            auth = psk
        }
        children {
            pok1-qz1-sr1-rk003-s17-10-66-3-35-ipv4 {
                local_ts = 11.6.81.0/24
                remote_ts = 11.0.209.0/24
                mode = tunnel
                inactivity = 130
                start_action = trap
                dpd_action = clear
                esp_proposals = aes256gcm8-sha256-modp2048
                rekey_time=60m
                rand_time=6m
                interface=ipsec0
                mark_in=5
            }
        }
        version = 2
        proposals = aes256gcm8-sha256-modp2048
        rekey_time=240m
        rand_time=24m
    }
}

**Comment by tobiasbrunner** [comment]
Completely depends on the use case/requirements.
